@xen-orchestra/xapi/vtpm.mjs

```diff
@@ -0,0 +1,34 @@
+import { incorrectState } from 'xo-common/api-errors.js'
+
+export default class Vtpm {
+  async create(vmRef) {
```

Comment:
Create methods should take a record-like object, if some params be explicitly passed, use a second param, take a look at `VM_create`: https://github.com/vatesfr/xen-orchestra/blob/b13302ddebe3bc653fbe1b7f579dd5bfe7ae2738/%40xen-orchestra/xapi/vm.mjs#L253-L326

```suggestion
  async create({ is_unique, VM: vmRef }) {
```

Reply:
I think the constructor should  take all the values accepted by the xapi, so the is_unique should be in the constructor

Reply:
Isn't it what I said? :thinking:

Reply:
I agreed with you, but with a little more than a smiley. 
Also I discussed with @MathieuRA of what exactly means is_unique, and what about the other settings available , thus reducing a lot this comment